src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -702,7 +703,11 @@ private function validateAndDenormalize(Type $type, string $currentClass, string
                 }
 
                 if ($collectionValueType) {
-                    $collectionValueBaseType = $collectionValueType instanceof UnionType ? $collectionValueType->asNonNullable()->getBaseType() : $collectionValueType->getBaseType();
+                    try {
+                        $collectionValueBaseType = $collectionValueType instanceof UnionType ? $collectionValueType->asNonNullable()->getBaseType() : $collectionValueType->getBaseType();
```

Comment:
```suggestion
                        $collectionValueBaseType = $collectionValueType instanceof UnionType ? $collectionValueType->getBaseType() : $collectionValueType->getBaseType();
```

Can we shorten it like this?

Reply:
@xabbuh Not sure what you mean, the left and right side of the expression would then be the same.

Reply:
Ah indeed, so we could even shorten this a bit more, right?

```php
try {
    $collectionValueBaseType = $collectionValueType->getBaseType();
} catch (TypeInfoLogicException) {
    $collectionValueBaseType = Type::mixed();
}
```

Reply:
If I understand correctly, the point of `asNonNullable()` is to filter out nullable types, eg if you have `int|null`, the base type if `int`. In that case it makes sense.

Reply:
The `UnionType` class already takes care of that: https://github.com/symfony/symfony/blob/e02306ba638c32d296c06c424ab11c35421c1743/src/Symfony/Component/TypeInfo/Type/UnionType.php#L43

Reply:
Oh, you're right, I totally missed that.

Reply:
Done